RestaurantReservation.API/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,226 @@
+        /// <param name="pageSize">The number of items per page.</param>
+        /// <returns>A list of employee DTOs.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult<IEnumerable<EmployeeDto>>> GetEmployees(int pageNumber = 1, int pageSize = 10)
+        {
+            if (pageNumber < 1 || pageSize < 1)
+                return BadRequest($"'{nameof(pageNumber)}' and '{nameof(pageSize)}' must be greater than 0.");
+
+            pageSize = Math.Min(pageSize, MaxPageSize);
+
+            var (employees, paginationMetadata) = await _employeeRepository.GetAllAsync(
+                _ => true, 
+                pageNumber,
+                pageSize
+            );
+
+            Response.Headers.Add("X-Pagination", JsonSerializer.Serialize(paginationMetadata));
+
+            return Ok(_mapper.Map<IEnumerable<EmployeeDto>>(employees));
+        }
+
+
+        /// <summary>
+        /// Retrieves an employee by their ID.
+        /// </summary>
+        /// <param name="id">The ID of the employee to retrieve.</param>
+        /// <returns>The employee DTO if found; otherwise, a 404 Not Found response.</returns>
+        [HttpGet("{id}", Name = "GetEmployee")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<EmployeeDto>> GetEmployee(int id)
+        {
+            var employee = await _employeeRepository.GetByIdAsync(id);
+            if (employee == null)
+                return NotFound();
+
+            return Ok(_mapper.Map<EmployeeDto>(employee));
+        }
+
+        /// <summary>
+        /// Retrieves a list of managers.
+        /// </summary>
+        /// <returns>A list of employee DTOs representing managers.</returns>
+        [HttpGet("managers")]
+        public async Task<ActionResult<IEnumerable<EmployeeDto>>> GetManagers()
+        {
+            var employees = await _employeeRepository.ListManagersAsync();
+
+            return Ok(_mapper.Map<IEnumerable<EmployeeDto>>(employees));
+        }
+
+        /// <summary>
+        /// Calculates the average order amount handled by a specific employee.
+        /// </summary>
+        /// <param name="employeeId">The ID of the employee for whom to calculate the average order amount.</param>
+        /// <returns>A 200 OK response with the average order amount; or a 404 Not Found response if no orders exist for the specified employee.</returns>
+        [HttpGet("{employeeId}/average-order-amount")]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetAverageOrderAmount(int employeeId)
+        {
+            try
```

Comment:
do you think it's the responsibility of the controller to catch and map exceptions ? and why is it only this endpoint that has this ?

Reply:
i added the trycatch in this endpoint because i faced an issue during test and i wanted to handle potential error specificaly for this case, generally i understand that exception handling should be done at a higher level like in middleware to avoid repeating the same code in every controller method this was just a temporary solution during the testing phase